clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1789,6 +1791,33 @@ public void testProcessBackgroundEventsTimesOut() throws Exception {
         }
     }
 
+    /**
+     * Tests that calling {@link Thread#interrupt()} before {@link KafkaConsumer#poll(Duration)}
+     * causes {@link InterruptException} to be thrown.
+     */
+    @Test
+    public void testPollThrowsInterruptExceptionIfInterrupted() {
```

Comment:
Is it possible to merge this new test into https://github.com/apache/kafka/blob/trunk/clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java#L1263? If not, maybe we can remove the TODO.

Reply:
Thanks for the review.

KafkaConsumerTest is really in general very specific to the LegacyKafkaConsumer and the old consumer group protocol so I don't think merging this into that test is appropriate. But the TODO in that test is clearly incorrect. I will fix it.